tests/test_polycutoffcomposite.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2019 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# =============================================================================
+import unittest
+
+import dimod
+
+from dwave.system import PolyCutOffComposite
+
+
+class CutoffChecker(dimod.PolySampler):
+    def __init__(self, child_sampler, expected_poly):
+
+        self.child = child_sampler
+        self.poly = expected_poly
+
+    def sample_poly(self, poly, **parameters):
+        assert self.poly == poly, '{} != {}'.format(self.poly, poly)
+        return self.child.sample_poly(poly, **parameters)
+
+    def parameters(self):
+        return self.child.parameters()
+
+    def properties(self):
+        return self.child.properties()
+
+
+class TestConstruction(unittest.TestCase):
+    def test_instantiation_smoketest(self):
+        sampler = PolyCutOffComposite(dimod.HigherOrderComposite(dimod.ExactSolver()), 0)
+
+        self.assertTrue(hasattr(sampler, 'sample_poly'))
+        self.assertTrue(hasattr(sampler, 'sample_hising'))
+        self.assertTrue(hasattr(sampler, 'sample_hubo'))
+
+    def test_wrap_bqm(self):
+        with self.assertRaises(TypeError):
+            PolyCutOffComposite(dimod.ExactSolver(), -1)
+
+
+class TestSampleHising(unittest.TestCase):
+    def setUp(self):
+        self.child = dimod.HigherOrderComposite(dimod.ExactSolver())
+
+    def test_empty(self):
+        h = {}
+        J = {}
+        cutoff = 1
+        expected = dimod.BinaryPolynomial({}, dimod.SPIN)
+
+        checker = CutoffChecker(self.child, expected)
+        samples = PolyCutOffComposite(checker, cutoff).sample_hising(h, J)
+
+        self.assertEqual(samples.record.sample.shape[1], 0)  # no variables
+
+    def test_linear(self):
+        # they are all isolated
+        h = {'a': -1, 'b': .5}
+        J = {}
+        cutoff = 1
+
+        # we cannot check in this case because all variables are isolated
+        # this results in exactly one variable being sent to ExactSolver and
+        # we don't know which one it will be, so we just check the correctness
+        # of the output
+        samples = PolyCutOffComposite(self.child, cutoff).sample_hising(h, J)
```

Comment:
default cutoff comparison is lt, so wouldn't a:-1 pass that and get sent to the exact solver?

Reply:
We don't apply the cutoff to the linear biases, only the higher-order. Linear biases are only removed if they are isolated. Given that this is for the QPU, I couldn't think of any reason to zero-out small linear biases, at worst they will be below the noise level of the system.